### ISSUE TITLE
haxe.format.JsonParser doesn't error on some invalid JSONs

### ISSUE THREAD

**Issue by nadako** [HaxeFoundation/haxe#5691]
The following should throw an error, but instead prints `true`, `false`, `null` and `123`.

``` haxe
class Main {
    static function main() {
        trace(haxe.format.JsonParser.parse("trueeee"));
        trace(haxe.format.JsonParser.parse("falseah"));
        trace(haxe.format.JsonParser.parse("nullol"));
        trace(haxe.format.JsonParser.parse("123asd"));
    }
}
```

**Comment by nadako** [comment]
Originally found in https://github.com/nadako/hxjsonast/issues/6

**Comment by nadako** [comment]
I think it only fails on stand-alone values like this, since within array/object it's checked that a comma or closing brace should follow.

**Comment by nadako** [comment]
Obviously, it's the same with `haxe.format.JsonParser.parse('{} asdasd ')`.
I think the proper fix would be to call something like this instead of `parseRec` in the `parse` function:

```haxe
function doParse() : Dynamic {
	var result = parseRec();
	var c;
	while( !StringTools.isEof(c = nextChar()) ) {
		switch( c ) {
		case ' '.code, '\r'.code, '\n'.code, '\t'.code:
			// allow trailing whitespace
		default:
			invalidChar();
		}
	}
	return result;
}
```

**Comment by nadako** [comment]
closed by #6734
